test/stdlib/MultilineString.swift

```diff
@@ -0,0 +1,56 @@
+// RUN: %target-run-stdlib-swift | %FileCheck %s
+// REQUIRES: executable_test
+
+// RUN: %target-build-swift %s 2> %t.warnings.txt
+// RUN: %FileCheck -check-prefix=CHECK-WARNINGS %s < %t.warnings.txt
+
+import Swift
+
+// ===---------- Multiline --------===
+
+func delimit(_ str: String) -> String {
+  return "<\(str)>"
+}
+
+// CHECK: -1-
+print("-1-")
+// CHECK-NEXT: {{^}}<Six
+// CHECK-NEXT: {{^}}""Zeta"">
+print(delimit("""
+    Six
+    ""Zeta""
+    """
+))
+
+// CHECK: -2-
+print("-2-")
+// CHECK-NEXT: {{^}}<  Eight
+// CHECK-NEXT: {{^}}Iota>
+print(delimit("""
+    Eight
+  Iota
+  """
+))
+
+// CHECK: -3-
+print("-3-")
+// CHECK-NEXT: {{^}}<  Nine
+// CHECK-NEXT: {{^}}  Kappa>
+print(delimit("""
+    Nine
+    Kappa
+  """
+))
+
+// CHECK: -4-
+print("-4-")
+// CHECK-NEXT: {{^}}<    Ten
+// CHECK-NEXT: {{^}}    Lambda>
+print(delimit("""
+    Ten
+    Lambda
+"""))
+
+// ===---------- Done --------===
+// CHECK-NEXT: Done.
+print("Done.")
```

Comment:
Can you add more tests to ensure that the errors and warnings you have added behave as intended?

Some other ideas:

* Check that only one leading newline and one trailing newline is stripped.
* Check that all other newlines are preserved, including multiple consecutive ones.
* Check that manually escaped `\t`, `\r`, `\n`, `\\` are all possible and correctly lexed, including at the end of a line.
* Check that a manually escaped `\r\n` is not normalized to `\n`.
* Check that string literal interpolation works correctly.
* Check that invalid ragged leading indents trigger the expected error.
* Check that escaping `\"""` works correctly.

Also, I would like to see, for the purposes of this particular implementation, that end-of-line newlines after `\` are preserved and that trailing whitespace is correctly preserved.